platform/security/certificate/security-certificate-keystoreeditor/src/main/java/org/codice/ddf/security/certificate/keystore/editor/KeystoreEditor.java

```diff
@@ -270,13 +352,16 @@ private synchronized void addToStore(String alias, String keyPassword, String st
                 Enumeration<String> aliases = jks.aliases();
                 while (aliases.hasMoreElements()) {
                     String jksAlias = aliases.nextElement();
-                    if (jks.isKeyEntry(jksAlias)) {
-                        Key key = jks.getKey(jksAlias, keyPassword.toCharArray());
-                        Certificate[] certificateChain = jks.getCertificateChain(jksAlias);
-                        store.setKeyEntry(alias, key, keyPassword.toCharArray(), certificateChain);
-                    } else {
-                        Certificate certificate = jks.getCertificate(alias);
-                        store.setCertificateEntry(alias, certificate);
+                    if (alias.equals(jksAlias)) {
```

Comment:
This is going to break uploading a jks. We should be basically taking every single key in the JKS, but checking that any specified alias exists in it.